nix/machines/default.nix

```diff
@@ -0,0 +1,68 @@
+{
+  self,
+  lib,
+  ...
+}:
+let
+  allEnvironmentPaths = lib.filter (path: lib.hasPrefix "example-setup-" (builtins.baseNameOf path)) (
+    lib.attrValues (builtins.readDir ../test-environments)
+  );
+
+  systems = [
+    "x86_64-linux"
+    "aarch64-linux"
+    "x86_64-darwin"
+    "aarch64-darwin"
+  ];
```

Comment:
Are these all the systems we're going to test for? Can we source this from somewhere else, maybe sync it up with the global `flake-parts` `systems` list?